app/routes/circulars/circulars.server.ts

```diff
@@ -107,8 +107,68 @@ export async function search({
   totalItems: number
 }> {
   const client = await getSearch()
+  const isLuceneSyntax = query ? query.includes(':') : false
+  const sortOrder = query
+    ? { _score: 'desc' }
+    : { circularId: { order: 'desc' } }
 
   const [startTime, endTime] = getValidDates(startDate, endDate)
+  const esQuery = query
+    ? {
+        function_score: {
+          query: isLuceneSyntax
+            ? {
+                query_string: {
+                  query: `${query}~`,
+                  fuzziness: 'AUTO',
+                },
+              }
+            : {
+                bool: {
+                  must: [
+                    {
+                      query_string: {
+                        query: `*${query}*`,
+                        default_field: '*',
+                        fuzziness: 'AUTO',
+                      },
+                    },
```

Comment:
Why do we need to insert wildcards into the query string? Can we not accomplish what we want by using the [fuzziness options](https://opensearch.org/docs/latest/query-dsl/full-text/index/#fuzzy-query-options)?

Reply:
<img width="781" alt="Screenshot 2023-10-11 at 12 57 43 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/97721ddb-a627-4820-9083-14f7352dfb35">
This is removing wildcards. the above is without lucene syntax and has the exact same results with lucene syntax. 
compare to the images I commented above and you will see that with wildcards that returned several hits. I felt like that was a highly requested feature.

Reply:
<img width="721" alt="Screenshot 2023-10-11 at 1 02 07 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/3237e642-e477-46b1-a293-81ac46c46cd8">

Reply:
@Courey, the reason that "230802" doesn't match what you expect it to may be that it is being treated as a GCN Circular ID here:

https://github.com/nasa-gcn/gcn.nasa.gov/blob/2ad4607f922529d7fa2a2867c99e1d0e11467aff/app/routes/circulars._archive._index.tsx#L41

and here:

https://github.com/nasa-gcn/gcn.nasa.gov/blob/main/app/routes/circulars/circulars.server.ts#L251-L264

Reply:
how should people search for an event by numbers as requested without it assuming it's a circular id? Do people use the GCN circular id more than the event id? I think that the results that someone would expect if they typed 1234 in the search bar would be to get results that contain 1234, not circular 1234. But I also don't know how people typically use this, so the gcn circular id could be what people are searching for.

Reply:
also, searching for a year would bring up a circular then. like if I wanted to search for results in 2023 it would redirect me to circular 2023

Reply:
for example, the naming convention of GRBs make it really easy to search for all GRBs in august of this year. I can get all the circulars for august by searching 2308
<img width="779" alt="Screenshot 2023-10-13 at 1 41 07 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/2e547e57-3f97-4ac0-b889-4e78aeb0c6e3">

Reply:
> how should people search for an event by numbers as requested without it assuming it's a circular id? Do people use the GCN circular id more than the event id?

Without user testing, I do not know the answer to that.